### ISSUE TITLE
Problem with votes

### ISSUE THREAD

**Issue by SeruhioX** [BenCodez/VotingPlugin#239]
For some reason, some users are not receiving their votes

`[11:37:53] [Server thread/INFO]: [VotingPlugin] Recieved a vote from 'Minecraft-MP.com' by player 'gxlit0'!
[11:37:53] [Craft Scheduler Thread - 502/WARN]: [VotingPlugin] Player gxlit0 has not joined before, disregarding vote, set AllowUnjoined to true to prevent this`

But this user does exist

And for some reason the user is repeated in the database with different uuid, this user is not the only one to happen

https://i.gyazo.com/070ec118229e8a7fc0aa12e8cb8e87bc.png

**Comment by BenCodez** [comment]
I have no idea right now. Maybe one uuid is online, one offline uuid?

Don't have any context to your setup.

Player's need to have been on the server, not be in the database.

**Comment by SeruhioX** [comment]
Yes, I imagine that one is the uuid offline and another the online
I want to clarify, I am using "FastLogin", therefore some users have uuid premium and others do not

I just updated from 5.9, this did not happen before

I will return to the old version in the meantime, since there are quite a few people who did not receive the vote today

For some reason when they vote disconnected, it takes like they are uuid offline and this happens, all those who have failed are premium users

**Comment by BenCodez** [comment]
Not much I can do for you

On Aug 4, 2017 2:03 PM, "SeDark" <notifications@github.com> wrote:

Yes, I imagine that one is the uuid offline and another the online
I want to clarify, I am using "FastLogin", therefore some users have uuid
premium and others do not

I just updated from 5.9, this did not happen before

I will return to the old version in the meantime, since there are quite a
few people who did not receive the vote today

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/Ben12345rocks/VotingPlugin/issues/239#issuecomment-320315288>,
or mute the thread

**Comment by SeruhioX** [comment]
I do not understand, if in version 5.9 it worked, why nothing can be done?

**Comment by BenCodez** [comment]
That's what happens on offline server. I changed nothing to do with uuids.
It should work the same

**Comment by SeruhioX** [comment]
The problem doesn't have to do with offline UUIDs, if the server managed offline UUIDs only then VotingPlugin would assign the offline UUIDs to the players and everything would be fine.
 
While I do have an offline server, the thing is while using FastLogin which detects when premium accounts login and assigns their online UUID to them, when these premium accounts vote while being online VotingPlugin checks their name and I assume their UUID too and works fine.
 
This player is using a premium account and is also using his online UUID due to FastLogin
https://gyazo.com/779bc6849d147029aac28165c083e547
 
 
 
But when they vote while offline it says the user has never joined the server and it ignores his vote.
 
This player has a premium account too.
https://gyazo.com/ae1d2c38e573a123bea98962a7d18913
 
 
It then creates an offline UUID with that player's name and adds it to the database. I'm not sure why it says the user never connected when in this next image you can clearly see he has joined and voted before and that his online UUID had already been saved in the database.
 
https://gyazo.com/11ac53362c1276c46af814064f9d5dc2
 
 
So basically if this premium user "rumbaGames" logs on with his premium account and is using his onine UUID and then votes it will work fine but the next day after the server restart if he votes before logging in this problem will occur.
 
 
Is there some way where you could make sure the plugin checks the name of the player trying to vote and then check if his UUID was already in the database instead of just assuming he never joined and generating a random offline UUID.

**Comment by BenCodez** [comment]
This method gets the uuid for every voting player:
https://github.com/Ben12345rocks/AdvancedCore/blob/master/AdvancedCore/src/com/Ben12345rocks/AdvancedCore/Util/Misc/PlayerUtils.java#L124

The server must be giving the plugin different uuids. That is what happens you mess with them.

That is the only problem I can find right now.

**Comment by BenCodez** [comment]
How does this help?
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

**Comment by SeruhioX** [comment]
I just checked and it still doesn't work

**Comment by BenCodez** [comment]
Well, like I said, I use the server to get uuids and have never had a problem with it.

**Comment by SeruhioX** [comment]
So when someone votes while offline, why does it say "Player X has not joined before, disregarding vote" when it is clear he has connected and voted many times before and the db already has his UUID, isn't this an issue with the plugin?

**Comment by BenCodez** [comment]
Because I check with the server, not the db. And it's using offline uuid.

**Comment by SeruhioX** [comment]
So can you do something so it checks with the db or it checks both, currently this is the best plugin for what I'm looking for but if it can't save the premium player's UUIDs and check them later from the db then I can't keep using it along with my network's system.

**Comment by BenCodez** [comment]
This version checks with the database, not sure if I can do anything with the uuid thing, since it is a mix of online and offline uuids.
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

**Comment by SeruhioX** [comment]
Now it is working most of time since it grabs from the database the already present online UUID from the players, but sometimes it still creates the offline UUIDs and rewards it instead of the pre-existing ones(this makes us have to delete the offline UUIDs whenever they are created). Is there a way to prevent the plugin from creating offline UUIDs when one player already has his premium UUID in the db. Or maybe if this isn't possible, when the plugin receives a vote then check if the player voting has 2 UUIDs assigned to him and choose the premium/online one.

**Comment by BenCodez** [comment]
There really isn't much I can do. Mixing uuids is never a good idea. A uuid
is a uuid, not way to tell them apart from online/offline. Only way is to
look them up, but it fails most of the time

**Comment by SeruhioX** [comment]
Aren't there ways of checking if a UUID is offline or online like through mojang's API? Anyways if it's too much trouble I'm willing to offer money if you think it can be done since I really need the plugin to work with my system and honestly there aren't any better options out there.

**Comment by BenCodez** [comment]
Maybe talk to the author of the plugin that you are using that handles the
uuids, he might be able to do something about it

**Comment by SeruhioX** [comment]
The only thing the fastlogin plugin does is check if the player's name and account is premium and give him a premium/online UUID so other plugins that manage UUIDs (both online and offline) can interact with this user as if he were a premium account although the network is offline, I was hoping votingplugin could do the same, but the idea I'm getting from what we've discussed is that it can't do this?

**Comment by BenCodez** [comment]
When a player votes offline it checks with the server to get the uuid, and
is giving the offline uuid instead of the one that fast login should give.

**Comment by SeruhioX** [comment]
Fastlogin only checks if the user logging in to the network has a premium name and account and treats him as a premium user in an offline server and gives him the corresponding benefits(his premium account skin etc...), it has nothing to do with giving information to other plugins about wether the player has an offline or online UUID, that's the job of each plugin. 

When they receive info most advanced plugins check the username of the player that's interacting with the plugin and then check with mojang if that user has a premium account and online UUID or if that user account doesn't exist and so treat it as an offline UUID so they can manage both premium and non premium players, I have been asking if you could make this possible.

P.D. Fastlogin works in bungeecord and in the login lobby, it's not in the server where votingplugin is so they're unrelated.

**Comment by BenCodez** [comment]
I never see servers with mixed uuids, and don't think most plugins account
for that. What if the player with the offline uuid wants to vote? How can I
account for that? The server is the one getting the uuid for the player, I
think I linked the code showing that.  Is bungeecord on in spigot config?
That might be your problem.

I have one solution in mind but don't think it will work

**Comment by SeruhioX** [comment]
"I never see servers with mixed uuids"
- Basically every server with FastLogin which has over 13k downloads.

"What if the player with the offline uuid wants to vote"
- Well as you have said many times, the server is already giving the offline uuid so non premium users will have no problem.

"Is bungeecord on in spigot config?"
- Yes.

So right now if I'm not entirely mistaken the process is the following: Player votes while offline, server grabs the players username and offline uuid and sends this info to votingplugin to do the rest of the work.

What should happen is when the server sends the offline uuid and username to votingplugin, the plugin should then check with mojang's api if this username exists in their db to see if it has a premium account and therefore an online UUID, if it exists, then grab that online UUID and give out the rewards to that player.

And if it doesn't exist when checking mojang's api that means the player that voted is non-premium so it will be fine if it uses the offline uuid the server gave you initially to give the rewards to.

**Comment by BenCodez** [comment]
I added an option to have uuid lookups happen in the database.

https://github.com/Ben12345rocks/VotingPlugin/blob/5.9.5/VotingPlugin/Resources/Config.yml#L771

https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

See how it goes.

**Comment by BenCodez** [comment]
Just reuploaded the version, had an error.
https://www.dropbox.com/s/zyt6bfllm73n6wb/VotingPlugin.jar?dl=0

**Comment by SeruhioX** [comment]
For now it's working correctly with a slight delay in commands but the rest is fine.
